### PR TITLE
Cleanup old assist bases

### PR DESCRIPTION
**Pull request by ashmrtn**
Add helper functions and tests that find
old assist bases and add them to the set
of backups to garbage collect. This
allows us to remove backups that aren't
displayed to the user because they had
some errors during them

The more recent assist backups can still
be used for incremental backups so we
don't want to garbage collect all of
them

This still does not wire this code into
any existing corso function, just adds
additional cleanup logic

---

#### Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [x] :clock1: Yes, but in a later PR
- [ ] :no_entry: No

#### Type of change

- [x] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)

* #3217

#### Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/internal/kopia/cleanup_backups_test.go [hunk 1]
```diff
@@ -529,6 +625,173 @@ func (suite *BackupCleanupUnitSuite) TestCleanupOrphanedData() {
 			buffer:    24 * time.Hour,
 			expectErr: assert.NoError,
 		},
+		// Tests dealing with assist base cleanup.
+		{
+			name: "AssistBase NotYoungest InBufferTime Noops",
```
Comment (pandeyabs) [inline src/internal/kopia/cleanup_backups_test.go:630]: Given these tests are testing some complex code behavior, could you please add some comments for each test for more context?

Comment (pandeyabs) [reply src/internal/kopia/cleanup_backups_test.go:630]: Also, can we add a test which has both assist & merge bases? Just to establish the cleanup behavior that we are always going to keep the most recent assist base, even if it's superseded by a merge base.

**Review by pandeyabs — APPROVED** [review]
LGTM! This is very well thought out!

File: src/internal/kopia/cleanup_backups.go [hunk 2]
```diff
@@ -114,8 +126,11 @@ func cleanupOrphanedData(
 		return clues.Wrap(err, "getting all backup models")
 	}
 
-	toDelete := maps.Clone(deets)
-	maps.Copy(toDelete, dataSnaps)
+	// assistBackups is the set of backups that have a
+	//   * a label denoting their an assist backup
```
[the remark this answers is not in the document]
Comment (ashmrtn) [reply src/internal/kopia/cleanup_backups.go]: oh whoops, guess I should proof read things more often

File: src/internal/kopia/cleanup_backups_test.go (hunk 1 above)
Comment (ashmrtn) [reply src/internal/kopia/cleanup_backups_test.go:634]: >can we add a test which has both assist & merge bases

Good callout! I've added a new test for that and added comments to all the assist base tests

**Pull request merged**